src/Symfony/Component/Filesystem/Watcher/INotifyWatcher.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Filesystem\Watcher;
+
+use Symfony\Component\Filesystem\Exception\IOException;
+
+/**
+ * @internal
+ */
+class INotifyWatcher implements WatcherInterface
+{
+    public function watch($path, callable $callback, float $timeout = null)
```

Comment:
What about if I would like to watch `path` recursively (like in the #32764 use case) ?